packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1076,13 +1089,87 @@ class RenderParagraph extends RenderBox
           buffer.write(label);
           offset += label.length;
         }
-        _cachedAttributedLabel = AttributedString(buffer.toString(), attributes: attributes);
+        _cachedAttributedLabels = <AttributedString>[AttributedString(buffer.toString(), attributes: attributes)];
       }
-      config.attributedLabel = _cachedAttributedLabel!;
+      config.attributedLabel = _cachedAttributedLabels![0];
       config.textDirection = textDirection;
     }
   }
 
+  ChildSemanticsConfigurationsResult _childSemanticsConfigurationsDelegate(List<SemanticsConfiguration> childConfigs) {
```

Comment:
The logic in this method is hard to follow and reason about with its many levels of nesting and multiple loops some of which short-circuit with `continue`. Any chance this can be refactored a bit (maybe broken out into individual methods or - if all else fails - with some comments to make this more maintainable?